### PR TITLE
fix cookbook for Chef 14 - prefix new_resource

### PR DESCRIPTION
**Pull request by jtimberman**
https://docs.chef.io/release_notes.html#breaking-changes

> Property names not using new_resource.NAME

### CODE REVIEW

**Comment by bdangit** [comment]
Hey @jtimberman #166 should fix the build environment and allow this one to go green as long as you rebase.  However, once this gets merged in, I won’t be able to do a release until after the weekend.

**Comment by bdangit** [comment]
Odd. Everyone should have been able to see.

```
/pipeline/source/.rubocop.yml: Style/SpaceBeforeFirstArg has the wrong namespace - should be Layout
RuboCop failed!
Running RuboCop...
Inspecting 27 files
.................W.........

Offenses:

resources/user.rb:28:39: W: Unused block argument - permission. You can omit the argument if you don't care about it.
    new_resource.permissions.each do |permission|
                                      ^^^^^^^^^^
resources/user.rb:37:35: W: Unused block argument - db. You can omit the argument if you don't care about it.
  new_resource.databases.each do |db|
                                  ^^
resources/user.rb:38:39: W: Unused block argument - permission. You can omit the argument if you don't care about it.
    new_resource.permissions.each do |permission|
                                      ^^^^^^^^^^

27 files inspected, 3 offenses
```

**Comment by bdangit** [comment]
Btw, you can run the following with the latest chef-dk:

```
rake # Quick tests only (rubocop + minitest)
rake test:complete # All tests (rubocop + minitest + kitchen)
```

**Comment by fishnix** [comment]
I can't see it either @bdangit - getting a 401 (although not a very clear one!)

screen shot 2018-04-24 at 5 09 17 pm

**Comment by bdangit** [comment]
@fishnix et al, I think wercker wants you to login so you can see the logs.  very bad form.  I'm open to porting this over back to travis if this CI builder doesn't have this limitation.

File: resources/user.rb [hunk 1]
```diff
@@ -17,35 +17,35 @@
...
+  new_resource.databases.each do |db|
+    unless client.list_users.map { |x| x['username'] || x['name'] }.member?(new_resource.username)
+      client.create_database_user(db, new_resource.username, new_resource.password)
+      new_resource.updated_by_last_action true
     end
-    permissions.each do |permission|
-      client.grant_user_privileges(username, db, permission)
-      updated_by_last_action true
+    new_resource.permissions.each do |permission|
+      client.grant_user_privileges(new_resource.username, new_resource.db, new_resource.permission)
```
Comment (bdangit) [inline resources/user.rb]: I think you meant to do the following:

```ruby
new_resource.permissions.each do |permission|
  client.grant_user_privileges(new_resource.username, db, permission)
```

File: resources/user.rb [hunk 2]
```diff
@@ -17,35 +17,35 @@
...
 end
 
 action :update do
-  client.update_user_password(username, password) if password
-  databases.each do |db|
-    permissions.each do |permission|
-      client.grant_user_privileges(username, db, permission)
+  client.update_user_password(new_resource.username, new_resource.password) if new_resource.password
+  new_resource.databases.each do |db|
+    new_resource.permissions.each do |permission|
```
Comment (bdangit) [inline resources/user.rb]:
```ruby
new_resource.permissions.each do |permission|
  client.grant_user_privileges(new_resource.username, db, permission)
```

**Comment by bdangit** [comment]
hey @jtimberman, I left some comments in a github review.  Let me know if you need help to finish this off.

**Comment by jtimberman** [comment]
@bdangit Thanks!

**Comment by bdangit** [comment]
@jtimberman the new error I see now is:

```
Running RuboCop...
Inspecting 27 files
...........................

27 files inspected, no offenses detected
Starting Foodcritic linting...
Checking 16 files
.........xxx.x.x
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/admin.rb:25
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/admin.rb:34
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/admin.rb:50
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/continuous_query.rb:31
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/continuous_query.rb:37
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/database.rb:19
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/database.rb:24
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/retention_policy.rb:24
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/retention_policy.rb:28
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/retention_policy.rb:35
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/user.rb:26
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/user.rb:30
FC085: Resource using new_resource.updated_by_last_action to converge resource: /pipeline/source/resources/user.rb:48
```

Keep in mind this is using the Chef docker image you gave me.

**Comment by kri5** [comment]
Any news on this?

It seems that you should not call `new_resource.updated_by_last_action` but use `use_inline_resources` in the resource file instead (cf http://www.foodcritic.io/#FC085)

**Comment by kri5** [comment]
Well it seems that `use_inline_resources` is not needed since Chef13, so the use of it might no even be needed, depending on which chef version this cookbook supports.

**Comment by bdangit** [comment]
fixed this in #168. going to accept this and merge in #168.

**Pull request merged**